javalib/src/main/scala/java/lang/process/UnixProcessGen2.scala

```diff
@@ -183,29 +190,10 @@ private[process] class UnixProcessHandleGen2(pidFd: CInt)(
     }
   }
 
-  private def osWaitForImpl(timeout: Option[Ptr[timespec]]): Boolean = try {
-    // caller should have returned before here if cachedExitValue is known.
-    if (hasExited) {
-      // Another waitFor() has reaped exitValue; nothing to do here.
-    } else if (LinktimeInfo.isLinux) {
-      linuxWaitForImpl(timeout)
-    } else if (LinktimeInfo.isMac || LinktimeInfo.isFreeBSD) {
-      bsdWaitForImpl(timeout)
-    } else {
-      /* Should never get here. Earlier dispatch should have called
-       * UnixProcessGen1.
-       */
-      throw new IOException("unsuported Platform")
-    }
-    true
-  } catch {
-    case _: InterruptedException if !Thread.currentThread().isInterrupted() =>
-      false
-  }
-
   /* Linux - ppoll()
+   * Return true if the call shouldn't be tried again.
    */
-  private def linuxWaitForImpl(timeout: Option[Ptr[timespec]]): Unit = {
+  private def linuxWaitForImpl(timeout: Option[Ptr[timespec]]): Boolean = {
```

Comment:
Fond as I am of the fine names, I think the use of method
and its implementation have morphed away from a strong
association with the `waitFor` methods.

Now, you will ask, what would I suggest for a name?
Something with a 'Wait' or better, which does not 
imply the Monitor wait or an os `wait()`.  

This is, I believe, not directly a correctness issue today.
I do think that names which correspond to the code
will help devos who do not have this code branded
into their brain.